docs_src/dslx_std.md

```diff
@@ -1188,6 +1188,15 @@ index is used in a match expression (which will eagerly evaluate all of its
 arms), to prevent it from creating an error at simulation time if the value is
 ultimately discarded from the unselected match arm.
 
+#### `std::distinct`
+
+```dslx-snippet
+pub fn distinct<COUNT: u32, N: u32, S: bool>(items: xN[S][N][COUNT], valid: bool[COUNT]) -> bool
+```
+
+Returns whether all the `items` are distinct (i.e. there are no duplicate
+items) after the `valid` mask is applied.
```

Comment:
This could possibly use an example or two, to clarify what it means to apply the `valid` mask in this case.

Reply:
I duplicated a unit test into the doc -- I'm not sure it's generally wise. Ideally we'd be able to link to a searched piece of text in the stdlib but github isn't capable of doing that. Really we need a rustdoc like system for DSLX to make this proposition more sustainable. See also #1049